contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,356 @@
+9. If a primary partition becomes unhealthy, the node is tainted and all pods running in it will be evicted by default, unless they tolerate that taint. Kubelet’s behavior on a node with unhealthy primary partition is undefined. Cluster administrators are expected to fix unhealthy primary partitions on nodes. 
+
+### Bob runs batch workloads and is unsure of “storage” requirements
+
+1. Bob can create pods without any “storage” resource requirements. 
+```yaml
+apiVersion: v1
+kind: pod
+metadata:
+ name: foo
+ namespace: myns
+spec:
+ containers:
+  name: fooc
+ volumes:
+  name: myEmptyDir
+  emptyDir:   
+```
+2. His cluster administrator being aware of the issues with disk reclamation latencies has intelligently decided not to allow overcommitting primary partitions. The cluster administrator has installed a LimitRange to “myns” namespace that will set a default storage size. Note: A cluster administrator can also specify bust ranges and a host of other features supported by LimitRange for local storage.
+```yaml
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: mylimits
+spec:
+   - default:
+     storage-logs: 200Mi
+     Storage-overlay: 200Mi
+     type: Container
+   - default:
+     storage: 1Gi
+     type: Pod
+```
+3. The limit range will update the pod specification as follows:
+```yaml
+apiVersion: v1
+kind: pod
+metadata:
+ name: foo
+spec:
+ containers:
+  name: fooc
+  resources:
+   limits:
+    storage-logs: 200Mi
+    storage-overlay: 200Mi
+ volumes:
+  name: myEmptyDir
+  emptyDir:
+   capacity: 1Gi
+```
+4. Bob’s “foo” pod can use upto “200Mi” for its containers logs and writable layer each, and “1Gi” for its “myEmptyDir” volume. 
+5. If Bob’s pod “foo” exceeds the “default” storage limits and gets evicted, then Bob can set a minimum storage requirement for his containers and a higher “capacity” for his EmptyDir volumes.
+```yaml
+apiVersion: v1
+kind: pod
+metadata:
+ name: foo
+spec:
+ containers:
+  name: fooc
+  resources:
+   requests:
+    storage-logs: 500Mi
+    storage-overlay: 500Mi
+ volumes:
+  name: myEmptyDir
+  emptyDir:
+   capacity: 2Gi
+```
+6. Note that it is not possible to specify a minimum storage requirement for EmptyDir volumes because we intent to limit overcommitment of local storage due to high latency in reclaiming it from terminated pods. 
+
+### Alice manages a Database which needs access to “durable” and fast scratch space
+
+1. Cluster administrator provisions machines with local SSDs and brings up the cluster
+2. When a new node instance starts up, a addon DaemonSet discovers local “secondary” partitions which are mounted at a well known location and creates HostPath PVs for them if one doesn’t exist already. The PVs will include a path to the secondary device mount points and include additional labels and annotations that help tie to a specific node and identify the storage medium.
```

Comment:
what is HostPath PV's ? I am guessing we are only introducing one local disk based PV's. Can we consistently call them LocalDisk PV ?